test/e2e/gcs_test.go

```diff
@@ -67,12 +56,61 @@ func TestGCSMinimal(t *testing.T) {
 		t.Skip("skipping on non-GCP platform")
 	}
 
+	gcscfg, err := gcs.GetConfig(mockLister)
+	if err != nil {
+		t.Fatalf("error reading gcs config: %v", err)
+	}
+
+	authConfig := make(map[string]string)
+	if err := json.Unmarshal([]byte(gcscfg.KeyfileData), &authConfig); err != nil {
+		t.Fatal("unable to unmarshal gcp auth config")
+	}
+
+	// by adding some salt in here we can check later on if the information
+	// got propagated from ImageRegistryPrivateConfigurationUser and into
+	// ImageRegistryPrivateConfiguration.
+	authConfig["tainted"] = "some extra salt"
+
+	taintedAuth, err := json.Marshal(authConfig)
+	if err != nil {
+		t.Fatal("unable to marshal gcp auth config")
+	}
+
+	// create a GCS bucket manually here so we can configure later on the
+	// registry to use it.
+	credentials, err := goauth2.CredentialsFromJSON(ctx, []byte(gcscfg.KeyfileData), gstorage.ScopeFullControl)
+	if err != nil {
+		t.Fatalf("error creating gcs credentials: %v", err)
+	}
+
+	gcli, err := gstorage.NewClient(ctx, goption.WithCredentials(credentials))
+	if err != nil {
+		t.Fatalf("error creating gcs client: %v", err)
+	}
+
+	bucketName := fmt.Sprintf("registry-test-%s", uuid.New().String())
```

Comment:
It's better to prefix it with ClusterID aka infra.Status.InfrastructureName, in this case the installer will know that it's our cluster and protect our from leaks (in case of timeouts, OOMKilled, and other problems).

https://github.com/openshift/installer/blob/aecb9201459e3e92f12344e78aaaf03b6550fde7/pkg/destroy/gcp/bucket.go#L18